src/main/java/org/openrewrite/staticanalysis/RemoveUnreachableCodeVisitor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.staticanalysis;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.Statement;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+class RemoveUnreachableCodeVisitor extends JavaVisitor<ExecutionContext> {
+
+  @Override
+  public J visitBlock(J.Block block, ExecutionContext executionContext) {
+    block = (J.Block) super.visitBlock(block, executionContext);
+
+    List<Statement> statements = block.getStatements();
+    Optional<Integer> maybeFirstJumpIndex = findFirstJump(statements);
+    if (!maybeFirstJumpIndex.isPresent()) {
+      return block;
+    }
+    int firstJumpIndex = maybeFirstJumpIndex.get();
+
+    if (firstJumpIndex == statements.size() - 1) {
+      // Jump is at the end of the block, so nothing to do
+      return block;
+    }
+
+    List<Statement> newStatements =
+        ListUtils.flatMap(
```

Comment:
Note: The way `ListUtils.map()` is implemented, it allows the function to return a `null`, which indicates that the corresponding element should be removed from the list.

Reply:
But in this particular case I think it would be even easier with `statements.subList(0, firstJumpIndex)` (give or take a off-by-one error 😄 ).

Reply:
Good point! 
https://github.com/openrewrite/rewrite-static-analysis/pull/206

Reply:
Ah, good point. I initially didn't use `sublist` to avoid the potential for creating a new list if the jump was at the end of the block, and therefore the statements should remain the same (though I suppose it's possible most `List` implementations return themselves if the full list is requested with `sublist`). But once I handled that case separately with the above `if`, we know the statements are going to be trimmed at this point anyway.